### PR TITLE
File input button

### PR DESCRIPTION
**Pull request by Haz901**

### CODE REVIEW

File: tests/integration/components/file-input-test.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      'accept all video files by when type is video');
+
+    this.set('type', 'audio');
+    assert.equal($input.getAttribute('accept'), 'audio/*',
+      'accept all audio files by when type is audio');
+
+    this.set('accept', '.foo');
+    await render(hbs`{{file-input accept=accept}}`);
+    $input = find('input');
+    assert.equal($input.getAttribute('accept'), '.foo',
```
Comment (michaelw8) [inline tests/integration/components/file-input-test.js]: I think this should be a separate test: 'Can override default accept', something like that.

File: addon/styles/components/file-input.scss [hunk 2]
```diff
@@ -0,0 +1,43 @@
+.file-input {
+  display:    inline-grid;
+
+  input {
+    opacity:        0;
+    pointer-events: none;
+    position:       absolute;
+    z-index:        -1;
+  }
```
[the remark this answers is not in the document]
Comment (Haz901) [reply addon/styles/components/file-input.scss]: This is the standard ugly HTML file input, which is hidden to replace it with the button and styled file list.
`display:none` works as well, for some reason I thought the input wouldn't work anymore when doing that.

File: addon/components/file-input.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    change() {
+      let button   = this.element.querySelector('button');
+      let fileList = this.element.querySelector('input').files;
+      let files    = A();
+
+      button.blur();
+
+      for (let file of fileList) {
+        files.pushObject(file);
+      }
```
[the remark this answers is not in the document]
Comment (Haz901) [reply addon/components/file-input.js]: No, because a `fileList` is not an array...

File: tests/integration/components/paper/file-input-test.js [hunk 4]
```diff
@@ -0,0 +1,144 @@
+import { render, find }       from '@ember/test-helpers';
+
```
Comment (cbroeren) [inline tests/integration/components/paper/file-input-test.js]: Remove this empty line or the alignment of the `from` on the first line

File: tests/integration/components/paper/file-input-test.js [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+    let $input = find('input');
+    assert.equal($input.getAttribute('accept'), '*/*',
+      'accept all files by default');
+
+    this.set('type', 'foobar');
+    $input = find('input');
+
+    assert.equal($input.getAttribute('accept'), '*/*',
+      'accept all files by when type is not a regular file type');
```
Comment (cbroeren) [inline tests/integration/components/paper/file-input-test.js]: ~~by~~

File: tests/integration/components/paper/file-input-test.js [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+    this.set('type', 'image');
+    assert.equal($input.getAttribute('accept'), 'image/*',
+      'accept all image files by when type is image');
```
Comment (cbroeren) [inline tests/integration/components/paper/file-input-test.js]: ~~by~~

File: tests/integration/components/paper/file-input-test.js [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+    this.set('type', 'video');
+    assert.equal($input.getAttribute('accept'), 'video/*',
```
Comment (cbroeren) [inline tests/integration/components/paper/file-input-test.js]: ~~by~~

File: tests/integration/components/paper/file-input-test.js [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+      'accept all video files by when type is video');
+
+    this.set('type', 'audio');
+    assert.equal($input.getAttribute('accept'), 'audio/*',
+      'accept all audio files by when type is audio');
```
Comment (cbroeren) [inline tests/integration/components/paper/file-input-test.js]: ~~by~~

File: tests/integration/components/paper/file-input-test.js [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+  test('it renders', async function(assert) {
+    await render(hbs`{{paper/file-input}}`);
+    assert.ok(find('button'), 'shows a button');
+    assert.ok(find('button > i.fa.fa-upload'), 'shows default icon in button');
+    assert.ok(find('input[type=file]'), 'Has (hidden) file input');
+    assert.equal(this.element.textContent.trim(), 'No file selected',
+      'shows default message');
+  });
+
+  test('sets `accept` attribute on input', async function(assert) {
```
Comment (cbroeren) [inline tests/integration/components/paper/file-input-test.js]: Is it possibile to allow multiple file types? Just a question, don't have to implement it for now.

File: tests/integration/components/paper/file-input-test.js [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+  });
+
+  test('accept attribute on input is overridden by component attribute',
+  async function(assert) {
+    this.set('type', 'image');
+    this.set('accept', '.foo');
+
+    await render(hbs`{{paper/file-input type=type accept=accept}}`);
+    assert.equal(find('input').getAttribute('accept'), '.foo',
+      'accept attribut is overridden by component attribute');
```
Comment (cbroeren) [inline tests/integration/components/paper/file-input-test.js]: *attribute*

File: addon/templates/components/file-input.hbs [hunk 11]
```diff
@@ -0,0 +1,32 @@
+{{#paper-button class="file-input__button" disabled=disabled iconButton=true raised=true onClick=(action "triggerInput")}}
+  {{component iconComponent iconName}}
+  {{#paper-tooltip}}
+    Select {{if multiple (pluralize type) (singularize type)}}
+  {{/paper-tooltip}}
+{{/paper-button}}
+
+{{#if fileCount}}
+  <label class="file-input__file-count">
```
Comment (cbroeren) [inline addon/templates/components/file-input.hbs]: Why did you put the file count in a `<label>`?

File: tests/integration/components/paper/file-input-test.js (hunk 9 above)
Comment (Haz901) [reply tests/integration/components/paper/file-input-test.js]: Yes it is, it works just the same as the normal `<input>` [accept attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#attr-accept)

**Pull request merged**